psyneulink/library/compositions/pytorchcomponents.py

```diff
@@ -0,0 +1,257 @@
+from psyneulink.core.components.functions.transferfunctions import Linear, Logistic, ReLU
+from psyneulink.library.compositions.pytorchllvmhelper import *
+from psyneulink.core import llvm as pnlvm
+
+try:
+    import torch
+    from torch import nn
+    torch_available = True
+except ImportError:
+    torch_available = False
+
+
+# helper method that identifies the type of function used by a node, gets the function
+# parameters and uses them to create a function object representing the function, then returns it
```

Comment:
it is required for the pytorch specific operators (e.g. torch.exp), which allows for pytorch to calculate the gradient from the computation graph.

Reply:
so pytorch can handle `pytorch.exp` but not `np.exp` of `**`?

Reply:
yes, I believe so. seems like it's an active effort (https://github.com/pytorch/pytorch/issues/2228)

Reply:
(pytorch/pytorch#2228) refers to introducing numpy like API in pytorch, it doesn't talk about supporting bumpy types and operations. what am I missing?

Reply:
see:https://github.com/pytorch/pytorch/issues/2228#issuecomment-488808650

Reply:
that talks about adding a compatible API rather than supporting operations from other packages.
using `torch.numpy` would add hard requirement on pytorch and break all configurations that are not supported by pytorch.

I'm ok with this transformation, just add a comment explaining that it's due to torch limitations.